client/fingerprint/network_unix.go

```diff
@@ -33,41 +32,34 @@ func NewNetworkFingerprinter(logger *log.Logger) Fingerprint {
 func (f *NetworkFingerprint) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
 	// newNetwork is populated and addded to the Nodes resources
 	newNetwork := &structs.NetworkResource{}
-	defaultDevice := ""
-	ip := ""
-
-	// 1. Use user-defined network device
-	// 2. Use first interface found in the system for non-dev mode. (dev mode uses lo by default.)
-	if cfg.NetworkInterface != "" {
-		defaultDevice = cfg.NetworkInterface
-		ip = f.ipAddress(defaultDevice)
-	} else {
+	var ip string
 
-		intfs, err := net.Interfaces()
-		if err != nil {
-			return false, err
-		}
+	interfaces, err := f.findInterfaces(cfg.NetworkInterface)
+	if err != nil {
+		f.logger.Println(fmt.Sprintf("[DEBUG] Error while detecting network interface during fingerprinting: %s", err.Error()))
```

Comment:
This should be f.logger.Printf() and then you can remove the Sprintf.
And following the [DEBUG] should be `[DEBUG] fingerprint.network: ...`
Here and below